src/test/java/catan/controllers/testcases/game/FinishGameTest.java

```diff
@@ -143,4 +94,123 @@ public void should_successfully_finish_game_when_target_victory_points_is_4_and_
+                .loginUser(USER_NAME_3, USER_PASSWORD_3)
+
+                .setHex(HexType.STONE, 11).atCoordinates(0, -2)
+                .setHex(HexType.BRICK, 2).atCoordinates(1, -2)
+                .setHex(HexType.WOOD, 2).atCoordinates(2, -2)
+
+                .setHex(HexType.STONE, 11).atCoordinates(-1, -1)
+                .setHex(HexType.WHEAT, 3).atCoordinates(0, -1)
+                .setHex(HexType.SHEEP, 3).atCoordinates(1, -1)
+                .setHex(HexType.BRICK, 4).atCoordinates(2, -1)
+
+                .setHex(HexType.STONE, 11).atCoordinates(-2, 0)
+                .setHex(HexType.WHEAT, 5).atCoordinates(-1, 0)
+                .setHex(HexType.EMPTY, null).atCoordinates(0, 0)
+                .setHex(HexType.WHEAT, 8).atCoordinates(1, 0)
+                .setHex(HexType.WOOD, 4).atCoordinates(2, 0)
+
+                .setHex(HexType.SHEEP, 9).atCoordinates(-2, 1)
+                .setHex(HexType.SHEEP, 5).atCoordinates(-1, 1)
+                .setHex(HexType.SHEEP, 8).atCoordinates(0, 1)
+                .setHex(HexType.WOOD, 6).atCoordinates(1, 1)
+
+                .setHex(HexType.WOOD, 10).atCoordinates(-2, 2)
+                .setHex(HexType.WHEAT, 2).atCoordinates(-1, 2)
+                .setHex(HexType.BRICK, 6).atCoordinates(0, 2)
+
+                .createNewPublicGameByUser(USER_NAME_1, targetVictoryPoints)
+                .joinPublicGame(USER_NAME_2)
+                .joinPublicGame(USER_NAME_3)
+
+                        // take last player from the list each time, when pulling move order from the list to have order: 3, 2, 1
+                .nextRandomMoveOrderValues(asList(3, 2, 1))
+
+                .setUserReady(USER_NAME_1)
+                .setUserReady(USER_NAME_2)
+                .setUserReady(USER_NAME_3);
+    }
+
+    private Scenario playPreparationStage() {
+        return scenario
+                .BUILD_SETTLEMENT(1).atNode(2, -2, "topLeft")
+                .BUILD_ROAD(1).atEdge(2, -2, "topLeft")
+                .END_TURN(1)
+
+                .BUILD_SETTLEMENT(2).atNode(2, -1, "bottomRight")
+                .BUILD_ROAD(2).atEdge(2, -1, "bottomRight")
+                .END_TURN(2)
+
+                .BUILD_SETTLEMENT(3).atNode(0, 2, "topRight")
+                .BUILD_ROAD(3).atEdge(0, 2, "topRight")
+                .END_TURN(3)
+
+                .BUILD_SETTLEMENT(3).atNode(0, 0, "bottomRight")
+                .BUILD_ROAD(3).atEdge(0, 0, "bottomRight")
+                .END_TURN(3)
+
+                .BUILD_SETTLEMENT(2).atNode(0, 0, "bottomLeft")
+                .BUILD_ROAD(2).atEdge(0, 0, "bottomLeft")
+                .END_TURN(2)
+
+                .BUILD_SETTLEMENT(1).atNode(0, 0, "top")
+                .BUILD_ROAD(1).atEdge(0, 0, "topLeft")
+                .END_TURN(1);
+    }
+
+    private Scenario giveResourcesToPlayerForRoadBuilding(int moveOrder) {
+        return scenario
+                .nextRandomDiceValues(asList(moveOrder, moveOrder, 6, 6, 6, 6))
```

Comment:
it is easier to read if you set next random values just before dice is thrown, each time when it is needed